lib/test.js

```diff
@@ -93,7 +93,27 @@ Test.prototype.run = function () {
         this.timeoutAfter(this._timeout);
     }
     this.emit('prerun');
-    this._cb(this);
+    var callbackReturn = this._cb(this);
+
+    if (
+        typeof Promise === 'function' &&
+        callbackReturn &&
+        typeof callbackReturn.then === 'function'
```

Comment:
why would we only support thenables if promises are available? This seems like it would make it a half-assed thenables implementation, either we support them 100% or not at all?

Reply:
because if they’re not available, we’d have to ship an entire Promises shim to be able to provide spec-compliant behavior.

I think it’s reasonable to gate this functionality behind the almost-certainty that Promise is being shimmed in the final application.